Modules/CategoryReference/classes/class.ilObjCategoryReferenceListGUI.php

```diff
@@ -111,9 +111,9 @@ public function initItem(
         $target_title = ilContainerReference::_lookupTitle($obj_id);
         $target_description = ilObject::_lookupDescription($target_obj_id);
 
-        $this->deleted = $tree->isDeleted($target_ref_id);
+        $this->deleted = !$target_ref_id || $tree->isDeleted($target_ref_id);
```

Comment:
@mjansenDatabay I am wondering, if this line is correct. Are there other cases where we got not target ref id disregarding the object deletion. But maybe we just go with the assumption that it is ok to treat it like the deletion case.

I think a trait would be the best solution for this.

Reply:
@alex40724

This is most probably caused by deleted target objects.

AFAIR there are existing mechanisms (relying on event handling) to prevent these orphaned database references in the `ilContainerReference` object types, but it's PHP (means crashes can easily occur depending on memory limits, max. exec. time, error-causing plugins, etc.) and we don't use a RI.

This PR is only meant to be a workaround. Better options could be: Providing a cron job which regularly deletes orphaned data (if existing).

The line itself is IMO correct. If the `$target_ref_id` cannot be evaluated as a boolean `true`, it can be considered (IMO) as `deleted/missing`.